php/libraries/NDB_BVL_Battery.class.inc

```diff
@@ -193,11 +193,19 @@ class NDB_BVL_Battery
         // generate a commentID
         $commentID = $this->_createCommentID($testName);
 
+        // generate the double data entry commentid
+        $ddeCommentID = 'DDE_'.$commentID;
+
         // instantiate instrument object to get table name attribute
         $obj = NDB_BVL_Instrument::factory($testName, '', '');
 
-        // insert into the test table
-        $DB->insert($obj->table, array('CommentID' => $commentID));
+        if ($obj->jsonData !== true) {
```

Comment:
JSON and DDE are not related. Why are you making DDE conditional on not saving the data as JSON?

Reply:
I think because the instrument table does not exist in a JSON data instrument, so you cant really insert anything into it even DDE rows